dbt_platform_helper/providers/semantic_version.py

```diff
@@ -84,20 +87,13 @@ def warn(self):
         pass
 
 
+@dataclass
 class PlatformHelperVersionStatus(VersionStatus):
-    def __init__(
-        self,
-        local: SemanticVersion = None,
-        latest: SemanticVersion = None,
-        deprecated_version_file: SemanticVersion = None,
-        platform_config_default: SemanticVersion = None,
-        pipeline_overrides: dict[str, str] = None,
-    ):
-        self.local = local
-        self.latest = latest
-        self.deprecated_version_file = deprecated_version_file
-        self.platform_config_default = platform_config_default
-        self.pipeline_overrides = pipeline_overrides if pipeline_overrides else {}
+    local: Optional[SemanticVersion] = None
+    latest: Optional[SemanticVersion] = None
+    deprecated_version_file: Optional[SemanticVersion] = None
+    platform_config_default: Optional[SemanticVersion] = None
+    pipeline_overrides: Optional[Dict[str, str]] = None
```

Comment:
```suggestion
    pipeline_overrides: Optional[Dict[str, str]] = field(default_factory=dict)
```
Make the field default to an empty dictionary like old version.
Add to the top `from dataclasses import field`